Sources/SecureXPC/Server/XPCServer.swift

```diff
@@ -177,6 +177,36 @@ public class XPCServer {
     // Routes
     private var routes = [XPCRoute : XPCHandler]()
     
+    /// Array of weak references to connections, used to update their dispatch queues
+    internal var connections = [WeakConnection]()
+    
+    /// Weak reference wrapper around an `xpc_connection_t`
+    internal class WeakConnection {
+        weak var connection: xpc_connection_t?
+        
+        init(_ connection: xpc_connection_t) {
+            self.connection = connection
+        }
+    }
+    
+    // Actual dispatch queue
+    private var _targetQueue: DispatchQueue? = nil
+    
+    /// The queue used to run the handlers associated with registered routes.
+    ///
+    /// Applying the target queue is asynchronous and non-preemptive and therefore will not interrupt the execution of an already-running handler. The queue
+    /// returned from reading this property will always be the one most recently set even if it is not yet the queue used to run handlers for all incoming requests.
+    public var targetQueue: DispatchQueue? {
+        set {
+            connections.compactMap{ $0.connection }.forEach{ xpc_connection_set_target_queue($0, newValue) }
+            _targetQueue = newValue
+        }
```

Comment:
What's a use-case for setting the queue once the connection is already started?

I can't think of a motivating use-case personally, so I'm biased to think the complexity around it isn't worth it.

Reply:
Also, I think the `_targetQueue`/`targetQueue` pair can just be simplified to a single `targetQueue: DispatchQueue? { didSet { ... } }`.

Reply:
I don't have a motivating use case, and definitely agree it adds a bit of complexity.

What ultimately motivated this was consistency with setting an error handler and registering routes - both of which can be done after starting the server.

I originally tried several different designs, but liked all them less than this one:
1. A `targetQueue` property with documentation that it'd be ignored if the server was already running
2. A `setTargetQueue(...)` function call which threw an error if the server was already started
3. Having `start()` and `startAndBlock()` now optionally take in a target queue

They were all to some degree simpler to implement, but all made the API for an external user more inconsistent and therefore complicated.  Of the three options \# 3 had the fewest gotchas, but was most inconsistent while \# 1 had the most gotchas (as it fails silently) but was most consistent.

Reply:
What do you think about making the target-queue immutable by taking it as a parameter to the initializer.

There would be no need for a `setTargetQueue`, or public `targetQueue` property setter, and thus, no need to throw an error if already started.

Reply:
> What do you think about making the target-queue immutable by taking it as a parameter to the initializer.

I originally looked into that, but it's not feasible (or at least would be quite complicated). This is because users of the API don't actually call the initializers they just access them via the factory methods. For example `XPCServiceServer` has a singleton instance which is statically initialized.

Reply:
~Yeah, those factory methods would just pass on the parameters to the initializes. What's the issue?~

> For example `XPCServiceServer` has a singleton instance which is statically initialized.

Ah.